cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionConsumer.java

```diff
@@ -55,4 +56,13 @@
    * @param succeeded whether or not processing of the specified partitions was successful
    */
   void onFinish(List<? extends Partition> partitions, boolean succeeded);
+
+  /**
+   * Same as {@link #onFinish(List, boolean)}, but allows specifying {@link PartitionKey}s
+   * instead of {@link co.cask.cdap.api.dataset.lib.Partition}s.
+   *
+   * @param partitionKeys list of partition keys to mark as either succeeded or failed processing
+   * @param succeeded whether or not processing of the specified partitions was successful
+   */
+  void onFinishWithKeys(List<? extends PartitionKey> partitionKeys, boolean succeeded);
```

Comment:
They receive a List<PartitionDetail>, not a List<PartitionKey>.
In your test case, you use Lists.transform() to convert that into PartitionKey, which obviously returns a List. What if a user calls Iterables.transform()? Or does not use Guava, but his own code? What if they serialize it into some storage and deserialize it for onFinish()? I don't think it is harmful to accept  Iterable<PartitionKey>, or even Iterable<? extends PartitionKey>.